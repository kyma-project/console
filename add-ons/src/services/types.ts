import { Filters, ConfigurationLabels, Configuration } from '../types';

export interface FiltersService {
  activeFilters: Filters;
  setFilterLabel: (label: string) => void;
  removeFilterLabel: (label: string) => void;
  removeAllFilters: () => void;
  hasActiveLabel: (key: string, value: string) => boolean;
}

export interface LabelsService {
  getUniqLabels: ConfigurationLabels;
}

export interface ConfigurationsService {
  originalConfigs: Configuration[];
  filteredConfigs: Configuration[];
}

export enum SubscriptionType {
  ADD = 'ADD',
  UPDATE = 'UPDATE',
  DELETE = 'DELETE',
}
