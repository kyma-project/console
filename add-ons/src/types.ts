import React from 'react';

export type BackendModule = string;

export interface SecretRef {
  name: string;
  namespace: string;
}

export interface Repository {
  url: string;
  secretRef?: SecretRef;
}

export interface RepositoryAddon {
  name: string;
  status: string;
  version: string;
  reason: string;
  message: string;
}

export interface RepositoryStatus {
  url: string;
  status: string;
  reason: string;
  message: string;
  addons: RepositoryAddon[];
}

export interface ConfigurationStatus {
  phase: string;
  repositories: RepositoryStatus[];
}

export interface Configuration {
  name: string;
  urls: ConfigurationURL[];
  labels: ConfigurationLabels;
  repositories: Repository[];
  status: ConfigurationStatus;
}

export interface ConfigurationLabels {
  [key: string]: string;
}
export type ConfigurationURL = string;

export interface FilterLabels {
  [key: string]: string[];
}

export interface Filters {
  search: string;
  labels: FilterLabels;
}

export interface ActiveFiltersAction {
  payload: {
    key: string;
    value: string;
  };
  type: ActiveFiltersActionType;
}

export enum ActiveFiltersActionType {
  SET_SEARCH = 'SET_SEARCH',
  SET_LABEL = 'SET_LABEL',
  REMOVE_LABEL = 'REMOVE_LABEL',
  REMOVE_ALL_FILTERS = 'REMOVE_ALL_FILTERS',
}

export interface RemoveFiltersInterface {
  removeFilterLabel: (key: string, value: string) => void;
  removeAllFiltersLabels: () => void;
}

export interface FiltersLabelsInterface {
  uniqueLabels: FilterLabels;
  setFilterLabel: (key: string, value: string) => void;
  hasActiveLabel: (key: string, value: string) => boolean;
}

export interface ConfigurationsService
  extends FiltersLabelsInterface,
    RemoveFiltersInterface {}

export interface Notification {
  title: string;
  content: string | React.ReactNode;
  color?: string;
  icon?: string;
}
