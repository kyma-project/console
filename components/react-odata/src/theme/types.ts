export interface ThemeInterface {
  color?: string;
}
