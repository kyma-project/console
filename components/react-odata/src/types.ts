export interface Child {
  attributes: { [key: string]: string };
  name: string;
  value: string;
  children: Child[];
}
