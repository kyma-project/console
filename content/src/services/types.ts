import { ClusterAssetGroup } from '@kyma-project/common';

export interface ClusterAssetGroups {
  [group: string]: ClusterAssetGroup[];
}

export interface Navigation {
  [group: string]: NavigationItem[];
}

export interface NavigationItem {
  name: string;
  group: string;
  displayName: string;
}
