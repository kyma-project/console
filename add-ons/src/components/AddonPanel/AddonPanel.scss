.addon-collapsible-panel {
    > * {
        padding: 8px;
    }
}

.addon-panel__error {
    background: #ffebeb;
}
