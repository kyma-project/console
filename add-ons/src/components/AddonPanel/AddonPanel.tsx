import React, { useContext, useState } from 'react';
import CollapsiblePanel from '../CollapsiblePanel/CollapsiblePanel';
import { Configuration } from '../../types';
import { Button } from 'fundamental-react';
import AddonsConfigurationBadge from './../AddonConfigurationBadge/AddonConfigurationBadge';
import LabelDisplay from './../LabelDisplay/LabelDisplay';
import DeleteConfigurationModal from '../Modals/DeleteConfigurationModal/DeleteConfigurationModal.container';
import AddUrlModal from '../Modals/AddUrlModal/AddUrlModal.container';
import AddonTable from './AddonTable/AddonTable';
import { DEFAULT_CONFIGURATION } from './../../constants';
import './AddonPanel.scss';
import { MutationsService } from '../../services/Mutations.service';
import { NotificationsService } from '@kyma-project/common';

interface ResyncButtonProps {
  configurationName: string;
}

const ResyncButton: React.FunctionComponent<ResyncButtonProps> = ({
  configurationName,
}) => {
  const {
    resyncAddonsConfiguration: [resyncAddonsConfiguration],
  } = useContext(MutationsService);
  const { errorNotification } = useContext(NotificationsService);

  async function handleResync() {
    try {
      await resyncAddonsConfiguration({
        variables: {
          name: configurationName,
        },
      });
    } catch (e) {
      errorNotification({
        title: 'Error',
        content: `Error while resyncing configuration ${configurationName}.`,
      });
    }
  }

  return <Button glyph="refresh" option="light" onClick={handleResync} />;
};

interface AddonPanelProps {
  config: Configuration;
}

const AddonPanel: React.FunctionComponent<AddonPanelProps> = ({ config }) => {
  const isConfigManaged = !config.labels
    ? undefined
    : Object.keys(config.labels)
        .filter(label => label.endsWith('managed')) // find keys ending with 'managed' - DNS prefix can be omitted
        .some(label => config.labels[label] === 'true');

  const isConfigDefault = config.name === DEFAULT_CONFIGURATION;

  const [isOpen, setIsOpen] = useState(true);

  const panelTitle = isConfigDefault ? `${config.name} (default)` : config.name;

  const showAddUrlButton = !isConfigManaged && !isConfigDefault;

  const actions = (
    <>
      {showAddUrlButton && <AddUrlModal configurationName={config.name} />}
      <ResyncButton configurationName={config.name} />
      <DeleteConfigurationModal configurationName={config.name} />
    </>
  );

  const panelContent = (
    <>
      {config.labels && <LabelDisplay readonlyLabels={config.labels} />}
      <AddonTable config={config} />
    </>
  );

  const hasRepositories = config.repositories.length > 0;

  return (
    <CollapsiblePanel
      title={panelTitle}
      actions={actions}
      isOpen={hasRepositories && isOpen}
      setIsOpen={setIsOpen}
      collapseDisabled={!hasRepositories}
      additionalHeaderContent={
        <AddonsConfigurationBadge
          status={config.status.phase}
          className="fd-has-margin-left-s"
        />
      }
      className="fd-has-margin-bottom-s addon-collapsible-panel"
    >
      {panelContent}
    </CollapsiblePanel>
  );
};

export default AddonPanel;
