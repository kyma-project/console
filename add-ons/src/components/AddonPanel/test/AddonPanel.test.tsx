import React from 'react';
import AddonPanel from './../AddonPanel';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Configuration } from '../../../types';

const configMock: Configuration = {
  name: 'configuration',
  urls: [],
  labels: {},
  repositories: [],
  status: {
    phase: 'ready',
    repositories: [],
  },
};

describe('AddonPanel', () => {
  it('Renders with minimal props', () => {
    const component = shallow(<AddonPanel config={configMock} />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
