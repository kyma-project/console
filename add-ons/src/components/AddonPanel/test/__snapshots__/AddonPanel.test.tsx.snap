// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddonPanel Renders with minimal props 1`] = `
<CollapsiblePanel
  actions={
    <React.Fragment>
      <AddUrlModalContainer
        configurationName="configuration"
      />
      <ResyncButton
        configurationName="configuration"
      />
      <DeleteConfigurationModalContainer
        configurationName="configuration"
      />
    </React.Fragment>
  }
  additionalHeaderContent={
    <AddonConfigurationBadge
      className="fd-has-margin-left-s"
      status="ready"
    />
  }
  className="fd-has-margin-bottom-s addon-collapsible-panel"
  collapseDisabled={true}
  isOpen={false}
  setIsOpen={[Function]}
  title="configuration"
>
  <LabelDisplay
    readonlyLabels={Object {}}
  />
  <AddonTable
    config={
      Object {
        "labels": Object {},
        "name": "configuration",
        "repositories": Array [],
        "status": Object {
          "phase": "ready",
          "repositories": Array [],
        },
        "urls": Array [],
      }
    }
  />
</CollapsiblePanel>
`;
